### PR TITLE
Use IPPrefix.Valid in other IPPrefix methods.

### PR DESCRIPTION
**Pull request by danderson**
Adjust some tests to check behavior on invalid prefixes.

Signed-off-by: David Anderson <dave@natulte.net>

### CODE REVIEW

**Comment by danderson** [comment]
Caused a couple minor changes to inlining, but I'm treating those similar to depaware: updated the test to make us think about whether a particular outlining is bad or not, but not a showstopper.

File: netaddr.go [hunk 1]
```diff
@@ -1034,10 +1034,14 @@ func MustParseIPPrefix(s string) IPPrefix {
 // Masked returns p in its canonical form, with bits of p.IP not in p.Bits masked off.
 // If p is zero or otherwise invalid, Masked returns the zero value.
 func (p IPPrefix) Masked() IPPrefix {
-	if m, err := p.IP.Prefix(p.Bits); err == nil {
-		return m
+	if !p.Valid() {
```
Comment (bradfitz) [inline netaddr.go:1037]: why is this necessary? won't IP.Prefix return an error if it's invalid?

Comment (danderson) [reply netaddr.go]: Doh, so it does. Reverted.
